tests/IntervalsTest.php

```diff
@@ -0,0 +1,380 @@
+                        'start' => '>= 4.0.0.0-dev',
+                        'end' => '< '.PHP_INT_MAX.'.0.0.0',
+                    ),
+                ), 'devConstraints' => array()),
+                '^4.0 || >= 4.5'
+            ),
+            'overlapping intervals should be merged even if the first has no start' => array(
+                array('intervals' => array(
+                    array(
+                        'start' => '>= 0.0.0.0-dev',
+                        'end' => '< 6.0.0.0-dev',
+                    ),
+                ), 'devConstraints' => array()),
+                '>= 5,< 6 || < 5.3'
+            ),
+            'overlapping intervals representing everything should become *' => array(
+                self::INTERVAL_ANY,
+                '>= 5 || <= 5'
+            ),
+            'equal intervals should be merged' => array(
+                array('intervals' => array(
+                    array(
+                        'start' => '>= 1.0.0.0-dev',
+                        'end' => '< 2.0.0.0-dev',
+                    ),
+                ), 'devConstraints' => array()),
+                '^1.0 || ^1.0'
+            ),
+            'weird input order should still be a good result' => array(
+                array('intervals' => array(
+                    array(
+                        'start' => '>= 0.0.0.0-dev',
+                        'end' => '< 2.0.0.0-dev',
+                    ),
+                ), 'devConstraints' => array()),
+                '< 2.0 || < 1.2'
+            ),
+            'weird input order should still be a good result/matches everything' => array(
+                self::INTERVAL_ANY,
+                '< 2.0 || >= 1'
+            ),
+            'weird input order should still be a good result, conjunctive' => array(
+                array('intervals' => array(
+                    array(
+                        'start' => '>= 1.0.0.0-dev',
+                        'end' => '< 2.0.0.0-dev',
+                    ),
+                ), 'devConstraints' => array()),
+                '< 2.0, >= 1'
+            ),
+            'conjunctive constraints should be intersected' => array(
+                array('intervals' => array(
+                    array(
+                        'start' => '>= 1.2.0.0-dev',
+                        'end' => '< 2.0.0.0-dev',
+                    ),
+                ), 'devConstraints' => array()),
+                '^1.0, ^1.2'
+            ),
+            'conjunctive constraints should be intersected/2' => array(
+                array('intervals' => array(
+                    array(
+                        'start' => '>= 1.5.0.0-dev',
+                        'end' => '< 1.7.0.0-dev',
+                    ),
+                ), 'devConstraints' => array()),
+                '^1.0, ^1.2, 1.4 - 1.8, 1.5 - 1.6, 1.5 - 2'
+            ),
+            'conjunctive constraints should be intersected/not flattened by version parser' => array(
```

Comment:
What does "flattened" mean in this context?

Reply:
This tests exactly the same as the above, but the VersionParser takes all constraints and makes a single MultiConstraint out of them all as it's a conjunctive list.. So wanted to make sure the nested/"not flattened" multiconstraints also works.

Reply:
Oh I see, the way it was written I thought somehow "not flattened" was an alterntive way to say "intersected", guess you could replace the "/" with an "and", but all good ;-)